diff ... libdrm-2.4.71/exynos/exynos_drm.c
--- libdrm-2.4.71-ref/exynos/exynos_drm.c	Thu Apr 21 09:40:06 2016
+++ libdrm-2.4.71/exynos/exynos_drm.c	Sat Nov 19 21:07:12 2016
@@ -35,7 +35,7 @@
 #include <unistd.h>
 
 #include <sys/mman.h>
-#include <linux/stddef.h>
+#include <stddef.h>
 
 #include <xf86drm.h>
 
diff ... libdrm-2.4.71/exynos/exynos_fimg2d.c
--- libdrm-2.4.71-ref/exynos/exynos_fimg2d.c	Sun Jan 17 16:22:14 2016
+++ libdrm-2.4.71/exynos/exynos_fimg2d.c	Sat Nov 19 21:07:12 2016
@@ -21,7 +21,7 @@
 #include <assert.h>
 
 #include <sys/mman.h>
-#include <linux/stddef.h>
+#include <stddef.h>
 
 #include <xf86drm.h>
 
diff ... libdrm-2.4.71/tests/exynos/exynos_fimg2d_test.c
--- libdrm-2.4.71-ref/tests/exynos/exynos_fimg2d_test.c	Sun Jan 17 16:22:14 2016
+++ libdrm-2.4.71/tests/exynos/exynos_fimg2d_test.c	Sat Nov 19 21:07:12 2016
@@ -22,7 +22,7 @@
 #include <unistd.h>
 
 #include <sys/mman.h>
-#include <linux/stddef.h>
+#include <stddef.h>
 
 #include <xf86drm.h>
 #include <xf86drmMode.h>
