--- libdrm-2.4.73-1/libdrm.pc.in	Tue Dec 27 16:59:16 2016
+++ libdrm-2.4.73/libdrm.pc.in	Tue Dec 27 16:59:50 2016
@@ -7,4 +7,4 @@
 Description: Userspace interface to kernel DRM services
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ldrm
-Cflags: -I${includedir} -I${includedir}/libdrm
+Cflags: -I${includedir} -I${includedir}/drm
diff -ur libdrm-2.4.73-1/amdgpu/libdrm_amdgpu.pc.in libdrm-2.4.73/amdgpu/libdrm_amdgpu.pc.in
--- libdrm-2.4.73-1/amdgpu/libdrm_amdgpu.pc.in	2016-12-27 17:09:11.730815329 +0300
+++ libdrm-2.4.73/amdgpu/libdrm_amdgpu.pc.in	2016-12-27 17:55:20.325354029 +0300
@@ -7,5 +7,5 @@
 Description: Userspace interface to kernel DRM services for amdgpu
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ldrm_amdgpu
-Cflags: -I${includedir} -I${includedir}/libdrm
+Cflags: -I${includedir} -I${includedir}/drm
 Requires.private: libdrm
diff -ur libdrm-2.4.73-1/etnaviv/libdrm_etnaviv.pc.in libdrm-2.4.73/etnaviv/libdrm_etnaviv.pc.in
--- libdrm-2.4.73-1/etnaviv/libdrm_etnaviv.pc.in	2016-11-14 20:55:20.000000000 +0300
+++ libdrm-2.4.73/etnaviv/libdrm_etnaviv.pc.in	2016-12-27 17:55:20.328666152 +0300
@@ -7,5 +7,5 @@
 Description: Userspace interface to etnaviv kernel DRM services
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ldrm_etnaviv
-Cflags: -I${includedir} -I${includedir}/libdrm
+Cflags: -I${includedir} -I${includedir}/drm
 Requires.private: libdrm
diff -ur libdrm-2.4.73-1/exynos/libdrm_exynos.pc.in libdrm-2.4.73/exynos/libdrm_exynos.pc.in
--- libdrm-2.4.73-1/exynos/libdrm_exynos.pc.in	2016-12-27 17:09:11.731173266 +0300
+++ libdrm-2.4.73/exynos/libdrm_exynos.pc.in	2016-12-27 17:55:20.326625001 +0300
@@ -7,5 +7,5 @@
 Description: Userspace interface to exynos kernel DRM services
 Version: 0.7
 Libs: -L${libdir} -ldrm_exynos
-Cflags: -I${includedir} -I${includedir}/libdrm -I${includedir}/exynos
+Cflags: -I${includedir} -I${includedir}/drm -I${includedir}/exynos
 Requires.private: libdrm
diff -ur libdrm-2.4.73-1/freedreno/libdrm_freedreno.pc.in libdrm-2.4.73/freedreno/libdrm_freedreno.pc.in
--- libdrm-2.4.73-1/freedreno/libdrm_freedreno.pc.in	2016-12-27 17:09:11.731477053 +0300
+++ libdrm-2.4.73/freedreno/libdrm_freedreno.pc.in	2016-12-27 17:55:20.324167348 +0300
@@ -7,5 +7,5 @@
 Description: Userspace interface to freedreno kernel DRM services
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ldrm_freedreno
-Cflags: -I${includedir} -I${includedir}/libdrm -I${includedir}/freedreno
+Cflags: -I${includedir} -I${includedir}/drm -I${includedir}/freedreno
 Requires.private: libdrm
diff -ur libdrm-2.4.73-1/intel/libdrm_intel.pc.in libdrm-2.4.73/intel/libdrm_intel.pc.in
--- libdrm-2.4.73-1/intel/libdrm_intel.pc.in	2016-12-27 17:09:11.731784746 +0300
+++ libdrm-2.4.73/intel/libdrm_intel.pc.in	2016-12-27 17:55:20.324762872 +0300
@@ -8,4 +8,4 @@
 Version: @PACKAGE_VERSION@
 Requires: libdrm
 Libs: -L${libdir} -ldrm_intel
-Cflags: -I${includedir} -I${includedir}/libdrm
+Cflags: -I${includedir} -I${includedir}/drm
diff -ur libdrm-2.4.73-1/nouveau/libdrm_nouveau.pc.in libdrm-2.4.73/nouveau/libdrm_nouveau.pc.in
--- libdrm-2.4.73-1/nouveau/libdrm_nouveau.pc.in	2016-12-27 17:09:11.732787352 +0300
+++ libdrm-2.4.73/nouveau/libdrm_nouveau.pc.in	2016-12-27 17:55:20.326031658 +0300
@@ -7,5 +7,5 @@
 Description: Userspace interface to nouveau kernel DRM services
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ldrm_nouveau
-Cflags: -I${includedir} -I${includedir}/libdrm -I${includedir}/libdrm/nouveau
+Cflags: -I${includedir} -I${includedir}/drm -I${includedir}/drm/nouveau
 Requires.private: libdrm
diff -ur libdrm-2.4.73-1/omap/libdrm_omap.pc.in libdrm-2.4.73/omap/libdrm_omap.pc.in
--- libdrm-2.4.73-1/omap/libdrm_omap.pc.in	2016-12-27 17:09:11.733077210 +0300
+++ libdrm-2.4.73/omap/libdrm_omap.pc.in	2016-12-27 17:55:20.327481008 +0300
@@ -7,5 +7,5 @@
 Description: Userspace interface to omap kernel DRM services
 Version: 0.6
 Libs: -L${libdir} -ldrm_omap
-Cflags: -I${includedir} -I${includedir}/libdrm -I${includedir}/omap
+Cflags: -I${includedir} -I${includedir}/drm -I${includedir}/omap
 Requires.private: libdrm
diff -ur libdrm-2.4.73-1/radeon/libdrm_radeon.pc.in libdrm-2.4.73/radeon/libdrm_radeon.pc.in
--- libdrm-2.4.73-1/radeon/libdrm_radeon.pc.in	2016-12-27 17:09:11.733377696 +0300
+++ libdrm-2.4.73/radeon/libdrm_radeon.pc.in	2016-12-27 17:55:20.328067455 +0300
@@ -7,5 +7,5 @@
 Description: Userspace interface to kernel DRM services for radeon
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ldrm_radeon
-Cflags: -I${includedir} -I${includedir}/libdrm
+Cflags: -I${includedir} -I${includedir}/drm
 Requires.private: libdrm
diff -ur libdrm-2.4.73-1/tegra/libdrm_tegra.pc.in libdrm-2.4.73/tegra/libdrm_tegra.pc.in
--- libdrm-2.4.73-1/tegra/libdrm_tegra.pc.in	2016-12-27 17:09:11.733694570 +0300
+++ libdrm-2.4.73/tegra/libdrm_tegra.pc.in	2016-12-27 17:55:20.323446018 +0300
@@ -7,5 +7,5 @@
 Description: Userspace interface to Tegra kernel DRM services
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ldrm_tegra
-Cflags: -I${includedir} -I${includedir}/libdrm
+Cflags: -I${includedir} -I${includedir}/drm
 Requires.private: libdrm
