--- libdrm-2.4.109/intel/intel_bufmgr_gem.c.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/intel/intel_bufmgr_gem.c	2021-12-28 14:55:05.655173523 +0000
@@ -3357,13 +3357,24 @@
 static pthread_mutex_t bufmgr_list_mutex = PTHREAD_MUTEX_INITIALIZER;
 static drmMMListHead bufmgr_list = { &bufmgr_list, &bufmgr_list };
 
+static int
+compare_file_descriptor(int fd1, int fd2) {
+	struct stat stat1, stat2;
+	if (fstat (fd1, &stat1) < 0)
+		return -1;
+	if (fstat (fd2, &stat2) < 0)
+		return -1;
+	return (stat1.st_dev == stat2.st_dev) && (stat1.st_ino == stat2.st_ino);
+}
+
 static drm_intel_bufmgr_gem *
 drm_intel_bufmgr_gem_find(int fd)
 {
 	drm_intel_bufmgr_gem *bufmgr_gem;
 
 	DRMLISTFOREACHENTRY(bufmgr_gem, &bufmgr_list, managers) {
-		if (bufmgr_gem->fd == fd) {
+		if (bufmgr_gem->fd == fd ||
+		    compare_file_descriptor (bufmgr_gem->fd, fd)) {
 			atomic_inc(&bufmgr_gem->refcount);
 			return bufmgr_gem;
 		}
--- libdrm-2.4.109/intel/intel_decode.c.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/intel/intel_decode.c	2021-12-28 14:55:05.656127083 +0000
@@ -398,6 +398,7 @@
 		format = "1555";
 		break;
 	case 3:
+	default:	/* Keep GCC happy. */
 		format = "8888";
 		break;
 	}
