--- libdrm-2.4.109/amdgpu/libdrm_amdgpu.pc.in.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/amdgpu/libdrm_amdgpu.pc.in	2021-12-28 13:56:31.717174443 +0000
@@ -6,6 +6,6 @@
 Name: libdrm_amdgpu
 Description: Userspace interface to kernel DRM services for amdgpu
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ldrm_amdgpu
-Cflags: -I${includedir} -I${includedir}/libdrm
+Libs: -L${libdir} -R${libdir} -ldrm_amdgpu
+Cflags: -I${includedir} -I${includedir}/drm
 Requires.private: libdrm
--- libdrm-2.4.109/etnaviv/libdrm_etnaviv.pc.in.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/etnaviv/libdrm_etnaviv.pc.in	2021-12-28 13:56:31.717523043 +0000
@@ -7,5 +7,5 @@
 Description: Userspace interface to etnaviv kernel DRM services
 Version: @PACKAGE_VERSION@
 Libs: -L${libdir} -ldrm_etnaviv
-Cflags: -I${includedir} -I${includedir}/libdrm
+Cflags: -I${includedir} -I${includedir}/drm
 Requires.private: libdrm
--- libdrm-2.4.109/exynos/libdrm_exynos.pc.in.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/exynos/libdrm_exynos.pc.in	2021-12-28 13:56:31.717782111 +0000
@@ -6,6 +6,6 @@
 Name: libdrm_exynos
 Description: Userspace interface to exynos kernel DRM services
 Version: 0.7
-Libs: -L${libdir} -ldrm_exynos
-Cflags: -I${includedir} -I${includedir}/libdrm -I${includedir}/exynos
+Libs: -L${libdir} -R${libdir} -ldrm_exynos
+Cflags: -I${includedir} -I${includedir}/drm -I${includedir}/exynos
 Requires.private: libdrm
--- libdrm-2.4.109/freedreno/libdrm_freedreno.pc.in.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/freedreno/libdrm_freedreno.pc.in	2021-12-28 13:56:31.718045519 +0000
@@ -6,6 +6,6 @@
 Name: libdrm_freedreno
 Description: Userspace interface to freedreno kernel DRM services
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ldrm_freedreno
-Cflags: -I${includedir} -I${includedir}/libdrm -I${includedir}/freedreno
+Libs: -L${libdir} -R${libdir} -ldrm_freedreno
+Cflags: -I${includedir} -I${includedir}/drm -I${includedir}/freedreno
 Requires.private: libdrm
--- libdrm-2.4.109/intel/libdrm_intel.pc.in.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/intel/libdrm_intel.pc.in	2021-12-28 13:56:31.718309459 +0000
@@ -7,5 +7,5 @@
 Description: Userspace interface to intel kernel DRM services
 Version: @PACKAGE_VERSION@
 Requires: libdrm
-Libs: -L${libdir} -ldrm_intel
-Cflags: -I${includedir} -I${includedir}/libdrm
+Libs: -L${libdir} -R${libdir} -ldrm_intel
+Cflags: -I${includedir} -I${includedir}/drm
--- libdrm-2.4.109/libdrm.pc.in.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/libdrm.pc.in	2021-12-28 13:56:31.718562768 +0000
@@ -6,5 +6,5 @@
 Name: libdrm
 Description: Userspace interface to kernel DRM services
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ldrm
-Cflags: -I${includedir} -I${includedir}/libdrm
+Libs: -L${libdir} -R${libdir} -ldrm
+Cflags: -I${includedir} -I${includedir}/drm
--- libdrm-2.4.109/libkms/libkms.pc.in.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/libkms/libkms.pc.in	2021-12-28 13:56:31.718872073 +0000
@@ -6,6 +6,6 @@
 Name: libkms
 Description: Library that abstracts away the different mm interface for kernel drivers
 Version: 1.0.0
-Libs: -L${libdir} -lkms
+Libs: -L${libdir} -R${libdir} -lkms
 Cflags: -I${includedir}/libkms
 Requires.private: libdrm
--- libdrm-2.4.109/nouveau/libdrm_nouveau.pc.in.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/nouveau/libdrm_nouveau.pc.in	2021-12-28 13:56:31.719138480 +0000
@@ -6,6 +6,6 @@
 Name: libdrm_nouveau
 Description: Userspace interface to nouveau kernel DRM services
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ldrm_nouveau
-Cflags: -I${includedir} -I${includedir}/libdrm -I${includedir}/libdrm/nouveau
+Libs: -L${libdir} -R${libdir} -ldrm_nouveau
+Cflags: -I${includedir} -I${includedir}/drm -I${includedir}/drm/nouveau
 Requires.private: libdrm
--- libdrm-2.4.109/omap/libdrm_omap.pc.in.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/omap/libdrm_omap.pc.in	2021-12-28 13:56:31.719394584 +0000
@@ -6,6 +6,6 @@
 Name: libdrm_omap
 Description: Userspace interface to omap kernel DRM services
 Version: 0.6
-Libs: -L${libdir} -ldrm_omap
-Cflags: -I${includedir} -I${includedir}/libdrm -I${includedir}/omap
+Libs: -L${libdir} -R${libdir} -ldrm_omap
+Cflags: -I${includedir} -I${includedir}/drm -I${includedir}/omap
 Requires.private: libdrm
--- libdrm-2.4.109/radeon/libdrm_radeon.pc.in.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/radeon/libdrm_radeon.pc.in	2021-12-28 13:56:31.719662132 +0000
@@ -6,6 +6,6 @@
 Name: libdrm_radeon
 Description: Userspace interface to kernel DRM services for radeon
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ldrm_radeon
-Cflags: -I${includedir} -I${includedir}/libdrm
+Libs: -L${libdir} -R${libdir} -ldrm_radeon
+Cflags: -I${includedir} -I${includedir}/drm
 Requires.private: libdrm
--- libdrm-2.4.109/tegra/libdrm_tegra.pc.in.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/tegra/libdrm_tegra.pc.in	2021-12-28 13:56:31.719935522 +0000
@@ -6,6 +6,6 @@
 Name: libdrm_tegra
 Description: Userspace interface to Tegra kernel DRM services
 Version: @PACKAGE_VERSION@
-Libs: -L${libdir} -ldrm_tegra
-Cflags: -I${includedir} -I${includedir}/libdrm
+Libs: -L${libdir} -R${libdir} -ldrm_tegra
+Cflags: -I${includedir} -I${includedir}/drm
 Requires.private: libdrm
