--- libdrm-2.4.109/util_double_list.h.orig	2021-11-25 21:33:02.000000000 +0000
+++ libdrm-2.4.109/util_double_list.h	2021-12-28 13:56:31.722430676 +0000
@@ -110,7 +110,7 @@
 #ifndef container_of
 #define container_of(ptr, sample, member)				\
     (void *)((char *)(ptr)						\
-	     - ((char *)&((typeof(sample))0)->member))
+	     - ((char *)&((__typeof(sample))0)->member))
 #endif
 
 #define LIST_FOR_EACH_ENTRY(pos, head, member)				\
