diff --git a/xf86drmMode.c b/xf86drmMode.c
index c809c44..715f23a 100644
--- libdrm-2.4.64/xf86drmMode.c.~1~	2015-09-14 10:33:51.340507095 +0300
+++ libdrm-2.4.64/xf86drmMode.c	2015-09-14 10:36:09.881744836 +0300
@@ -850,7 +850,9 @@
 	drmClose(fd);
 	return 0;
 #endif
-	return -ENOSYS;
+
+/* for now return 0 on solaris */
+	return 0;
 }
 
 int drmModeCrtcGetGamma(int fd, uint32_t crtc_id, uint32_t size,
